src/state/vaultsLifecycle.ts

```diff
@@ -0,0 +1,429 @@
+      ]);
+
+      const parsedAccount = VaultAccountCalculator.getAccountVaultResponse(acc);
+      const parsedTransfers = VaultAccountCalculator.getTransfersBetweenResponse(transfers);
+
+      if (parsedAccount == null || parsedTransfers == null) {
+        return wrapNullable(undefined);
+      }
+      return wrapNullable(
+        VaultAccountCalculator.calculateUserVaultInfo(parsedAccount, parsedTransfers)
+      );
+    },
+    ...vaultQueryOptions,
+  });
+  return accountVault?.data;
+}).dispatchValue((dispatch, value) => {
+  dispatch(setVaultAccount(value));
+});
+
+function useDebounceHf<T>(value: T, delayMs?: number): T {
+  const [debouncedValue, setDebouncedValue] = useStateHf<T>(value);
+
+  useEffectHf(() => {
+    const timer = setTimeout(() => setDebouncedValue(value), delayMs ?? 500);
+
+    return () => {
+      clearTimeout(timer);
+    };
+  }, [value, delayMs]);
+
+  return debouncedValue;
+}
+
+const vaultFormAmountDebounced = createHookedSelector(
+  [(state) => state.vaults.vaultForm.amount],
+  (amount) => {
+    return useDebounceHf(amount, 500);
+  }
+);
+
+export const vaultFormSlippage = createHookedSelector(
+  [
+    vaultFormAmountDebounced,
+    (state) => state.vaults.vaultForm.operation,
+    (state) => state.vaults.vaultAccount,
+  ],
+  (amount, operation, vaultBalance) => {
+    const { data: slippageResp } = useQueryHf({
+      queryKey: ['vaultSlippage', amount, operation],
+      queryFn: async () => {
+        if (operation === 'DEPOSIT' || amount.trim() === '') {
+          return wrapNullable(undefined);
+        }
+        const sharesToWithdraw = MustBigNumber(amount).div(
+          // usdc / share to determine share value
+          MustBigNumber(vaultBalance?.balanceUsdc).div(MustBigNumber(vaultBalance?.balanceShares))
+        );
+        const slippage = await placeholderFetchVaultFormSlippage(
+          sharesToWithdraw.decimalPlaces(0, BigNumber.ROUND_FLOOR).toNumber(),
+          MustBigNumber(amount).times(0.96145).toNumber()
+        );
+
+        const parsedSlippage =
+          VaultDepositWithdrawFormValidator.getVaultDepositWithdrawSlippageResponse(slippage);
+
+        return wrapNullable(parsedSlippage);
+      },
+      ...vaultQueryOptions,
+    });
+    return slippageResp?.data;
+  }
+).dispatchValue((dispatch, value) => {
+  dispatch(setVaultFormSlippageResponse(value ?? undefined));
+});
+
+export const vaultFormValidation = createHookedSelector(
+  [
+    vaultFormAmountDebounced,
+    createAppSelector(
+      [
+        (state) => state.vaults.vaultForm.operation,
+        (state) => state.vaults.vaultForm.slippageAck,
+        (state) => state.vaults.vaultForm.confirmationStep,
+        (state) => state.vaults.vaultForm.slippageResponse,
+      ],
+      (operation, slippageAck, confirmationStep, slippageResponse) => ({
+        operation,
+        slippageAck,
+        confirmationStep,
+        slippageResponse,
+      })
+    ),
```

Comment:
nit: to improve readability, maybe this can be abstracted to `getVaultForm` from `vaultSelectors`

Reply:
Yeah I'll pull these selectors out for readability.